packages/flutter/lib/src/widgets/text.dart

```diff
@@ -201,6 +201,77 @@ class DefaultTextStyle extends InheritedTheme {
   }
 }
 
+/// The [TextHeightBehavior] that will apply to descendant [Text] widgets which
+/// have not explicitly set [Text.textHeightBehavior].
+///
+/// See also:
+///
+///  * [DefaultTextStyle], which defines a [TextStyle] to apply to descendant
+///    [Text] widgets.
+class DefaultTextHeightBehavior extends InheritedTheme {
+  /// Creates a default text height behavior for the given subtree.
+  ///
+  /// This will take precedence over [DefaultTextStyle.textHeightBehavior].
+  ///
+  /// The [textHeightBehavior] and [child] arguments are required and must not be null.
+  const DefaultTextHeightBehavior({
+    Key key,
+    @required this.textHeightBehavior,
+    @required Widget child,
+  }) :  assert(textHeightBehavior != null),
+        assert(child != null),
+        super(key: key, child: child);
+
+  /// A const-constructible default text height behavior that provides a
+  /// [TextHeightBehavior] of null.
+  ///
+  /// Returned from [of] when the given [BuildContext] doesn't have an enclosing
+  /// [DefaultTextHeightBehavior].
+  ///
+  /// This constructor creates a [DefaultTextStyle] that lacks a [child], which
+  /// means the constructed value cannot be incorporated into the tree.
+  const DefaultTextHeightBehavior.fallback({ Key key })
+      : textHeightBehavior = null,
+        super(key: key, child: null);
+
+  /// {@macro flutter.dart:ui.textHeightBehavior}
+  final ui.TextHeightBehavior textHeightBehavior;
+
+  /// The closest instance of this class that encloses the given context.
+  ///
+  /// If no such instance exists, returns an instance created by
+  /// [DefaultTextHeightBehavior.fallback], which contains fallback values.
+  ///
+  /// Typical usage is as follows:
+  ///
+  /// ```dart
+  /// DefaultTextHeightBehavior defaultTextHeightBehavior = DefaultTextHeightBehavior.of(context);
+  /// ```
+  static DefaultTextHeightBehavior of(BuildContext context) {
```

Comment:
Instead of returning the inherited widget, we should return textHeightBehavior and `const TextHeightBehavior()` by default

Reply:
Done, as discussed offline with the minor tweak of returning `null` by default.